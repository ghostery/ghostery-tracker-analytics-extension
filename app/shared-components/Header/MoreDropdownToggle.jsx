import React, { Component } from 'react';
import PropTypes from 'prop-types';

import './Header.scss';

// NOTE: This toggle component must be an extension of React.Component rather than
// a functional component because the Boostrap <Dropdown /> component requires that
// custom toggle and menu components be able to accept refs

// NOTE: I imagine we will end making a custom component for the settings dropdown
// menu as well, which works in the same way as the toggle (ie. as={CustomMenu})

class CustomMoreToggle extends Component {
  handleClick = (e) => {
    e.preventDefault();

    const { onClick } = this.props;
    onClick(e);
  }

  render() {
    return (
      <svg
        xmlns="http://www.w3.org/2000/svg"
        className="Header__dropdownToggle d-inline-block align-top"
        width="10"
        height="20"
        viewBox="0 0 4 16"
        onClick={this.handleClick}
      >
        <path d="M0 0h4v4H0zm0 6h4v4H0zm0 6h4v4H0z" fill="#FFF" fillRule="evenodd" />
      </svg>
    );
  }
}

CustomMoreToggle.propTypes = {
  onClick: PropTypes.func.isRequired,
};

export default CustomMoreToggle;
