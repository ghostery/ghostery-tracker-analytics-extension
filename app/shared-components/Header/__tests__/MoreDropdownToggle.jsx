import React from 'react';
import { shallow } from 'enzyme';
import chrome from 'sinon-chrome';
import path from 'path';
import fakeProps from 'react-fake-props';

import MoreDropdownToggle from '../MoreDropdownToggle';

jest.mock('../../../../src/classes/Globals', () => ({
  BROWSER_INFO: { displayName: 'Chrome' },
}));

describe('app/shared-components/Header/MoreDropdownToggle', () => {
  describe('Snapshot tests with shallow', () => {
    test('MoreDropdownToggle component should render', () => {
      global.chrome = chrome;
      const componentPath = path.join(__dirname, '../MoreDropdownToggle.jsx');
      const props = fakeProps(componentPath);

      const component = shallow(<MoreDropdownToggle {...props} />);
      expect(component).toMatchSnapshot();
    });
  });

  describe('Functionality testing', () => {
    test('handleClick function is called', () => {
      const componentPath = path.join(__dirname, '../MoreDropdownToggle.jsx');
      const props = fakeProps(componentPath);

      const component = shallow(<MoreDropdownToggle {...props} />);
      const handleClick = jest.fn();
      component.instance().handleClick = handleClick;
      component.instance().forceUpdate();

      component.find('.Header__dropdownToggle').simulate('click', { preventDefault: () => {} });
      expect(handleClick.mock.calls.length).toEqual(1);
    });
  });
});
