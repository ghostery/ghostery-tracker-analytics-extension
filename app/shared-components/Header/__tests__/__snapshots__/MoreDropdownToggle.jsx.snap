// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app/shared-components/Header/MoreDropdownToggle Snapshot tests with shallow MoreDropdownToggle component should render 1`] = `
<svg
  className="Header__dropdownToggle d-inline-block align-top"
  height="20"
  onClick={[Function]}
  viewBox="0 0 4 16"
  width="10"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M0 0h4v4H0zm0 6h4v4H0zm0 6h4v4H0z"
    fill="#FFF"
    fillRule="evenodd"
  />
</svg>
`;
